cfile/hogfile.cpp

```diff
@@ -408,30 +292,32 @@ int CreateNewHogFile(const char *hogname, int nfiles, const char **filenames, vo
       return HOGMAKER_COPY;
     }
     fclose(ifp);
-    // Setup the update message and send it
-    char msg[256];
-    int ipct = int(100.0 * (double(i) / double(header.nfiles)));
-    snprintf(msg, sizeof(msg), "Creating Hog File... (%d%% done)", ipct);
-    if (UpdateFunction != NULL)
+    if (UpdateFunction != nullptr) {
+      // Setup the update message and send it
+      char msg[256];
+      int ipct = int(100.0 * (double(i) / double(header.nfiles)));
+      snprintf(msg, sizeof(msg), "Creating Hog File... (%d%% done)", ipct);
       UpdateFunction(msg);
+    }
   }
   // now write the real index
   fseek(hog_fp, table_pos, SEEK_SET);
   for (i = 0; i < header.nfiles; i++) {
-    if (!WRITE_FILE_ENTRY(hog_fp, &table[i])) {
+    if (!WriteHogEntry(hog_fp, &table[i])) {
       delete[] table;
       fclose(hog_fp);
       strcpy(hogerr_filename, hogname);
       return HOGMAKER_OUTFILE;
     }
   }
-  //	cleanup
+  // cleanup
   fclose(hog_fp);
   delete[] table;
-  // Setup the update message and send it
-  char msg[256];
-  snprintf(msg, sizeof(msg), "Done Creating Hog File.");
-  if (UpdateFunction != NULL)
+  if (UpdateFunction != nullptr) {
+    // Setup the update message and send it
+    char msg[256];
+    snprintf(msg, sizeof(msg), "Done Creating Hog File.");
```

Comment:
I know this is old code, but... why `snprintf`? This can just simply be `const char* msg = "Done Creating Hog File."` unless I'm missing something :D.

Reply:
This message was used for callback function, so there no need to allocate/deallocate char * array on caller side. Or, maybe I'm missing something too :).

Reply:
What I'm saying is, this message can be a non-allocated string, just a compile time one instead. But no need to change that, it's just silly but doesn't break anything.